apps/scratch/src/hooks/useEmbeddedScratch.ts

```diff
@@ -33,12 +33,78 @@ export const useEmbeddedScratch = (
           break;
         case "getSubmission":
           if (vm) {
-            respondToMessageEvent({
-              procedure: "getSubmission",
-              result: new Blob([vm.toJSON()], {
-                type: "application/json",
-              }),
-            });
+            let assertionsEnabled = false;
+            const setAssertionsEnabled = (enabled: boolean): void => {
+              assertionsEnabled = enabled;
+            };
+
+            vm.runtime.once(
+              "ARE_ASSERTIONS_ENABLED_RESPONSE",
+              setAssertionsEnabled,
+            );
+
+            // check if assertions are enabled
+            vm.runtime.emit("ARE_ASSERTIONS_ENABLED_QUERY");
```

Comment:
if this whole thing does what I think it does, it deserves to be its own function to abstract away the arcane functioning of the Scratch VM. Something like `areAssertionsEnabled(vm)`.
I'd also add, within the function, inline comments pointing to `ARE_ASSERTIONS_ENABLED_QUERY` triggering a call to the `ARE_ASSERTIONS_ENABLED_RESPONSE` callback - I'm just assuming this is what it does :-)